modules/optimization/src/optimizationreporters/ParameterMeshOptimization.C

```diff
@@ -80,39 +131,128 @@ ParameterMeshOptimization::ParameterMeshOptimization(const InputParameters & par
   if (orders.size() > 1 && orders.size() != _nparams)
     paramError("parameter_orders",
                "There must be an order associated with each group of parameters.");
+
   // Size checks for input file parameter data
-  if (!initial_condition.empty() && initial_condition.size() != _nparams)
-    paramError("initial_condition",
+  if (!constant_group_initial_condition.empty() &&
+      constant_group_initial_condition.size() != _nparams)
+    paramError("constant_group_initial_condition",
                "There must be an initial condition associated with each group of parameters.");
```

Comment:
all these error messages could be clearer if you said "size does not match"

Reply:
I added output for the sizes.